src/Groups/directproducts.jl

```diff
@@ -53,7 +56,7 @@ The parameter `morphisms` is `false` by default. If it is set `true`, then the o
 function inner_direct_product(L::AbstractVector{T}; morphisms=false) where T<:Union{PcGroup,PermGroup,FPGroup,MatrixGroup}
    P = GAP.Globals.DirectProduct(GAP.julia_to_gap([G.X for G in L]))
    if T==MatrixGroup     # check that the matrix groups have the same base ring
-      length(Set([GAP.Globals.FieldOfMatrixGroup(H.X) for H in G.L]))==1 || throw(ArgumentError("The result is not a matrix group"))
+      length(Set([GAP.Globals.FieldOfMatrixGroup(H.X) for H in L]))==1 || throw(ArgumentError("The result is not a matrix group"))
```

Comment:
```suggestion
      length(Set(GAP.Globals.FieldOfMatrixGroup(H.X) for H in L))==1 || throw(ArgumentError("The result is not a matrix group"))
```
This should give the same result without allocating an intermediate array. If the eltype is known to be `T`, writing `Set{T}(...)` can also be advantageous in some cases.